### DISCUSSION TITLE
Ubuntu, QT, compiling Savvy CAN === a total and utter train wreck!

### DISCUSSION THREAD

**Discussion by daveHitchman** [collin80/SavvyCAN#728] [Q&A]
I guess that savvy can has been adjusted to use some more up to date QT than the default for Ubuntu - judging from the breaks in the APIs. (Though I think an early defect in canconnection about loadRelaxed is probably not QT related - not sure yet.
So I tried QTs instructions for downloading a later one, crash bang wallop, this doesnt work, that doesnt work, nothing works. 

Now I am certain that someone thinks it is great to be at the bleeding edge of some monstrosity like QT but please can someone update the Readme with actual functioning instructions that someone who doesnt want to be a guru in QT, doesnt want to be a guru in savvy can but just wants to use it to graph some captured can logs can use successfully? I do appreciate the masses of effort that goes into these things, it should save me a shed of time - I dont want to replicate all that effort - but it seems that in general with ALL the open source stuff I have encountered the ship sinks for want of a happorth of tar.

**Comment by collin80** [comment]
Believe me, I understand being frustrated. I hate that too. I have also found open source projects that I wanted to compile only to find that they take specific libraries I don't have and require such fiddly configuration that it is all but impossible for me to get the dumb thing to compile. So, I actually thought that SavvyCAN was reasonably easy to get compiling. That might sound laughable now but that was the goal. It doesn't have hardly any external dependencies other than downloading QT and being able to compile QT apps. But, you are correct, it has been updated somewhat recently to target newer versions of QT. Well, somewhat new... It doesn't (yet) require QT6 which has been out for multiple years (2 or 3 I think) but it does essentially require the very latest version of QT5 in order to work. I'm actually compiling it with the default QT version on Ubuntu but I'm on the bleeding edge Ubuntu so it is likely that the more stable Ubuntu versions don't have the needed QT version. Also, you're right, directly trying to download QT is a nightmare. You have to sign up for an account to get access and then there are a million options. The SavvyCAN repo actually automatically builds the latest version of the code so the releases always have a continuous integration build of the latest source code. This is done by this script: https://github.com/collin80/SavvyCAN/blob/master/.github/workflows/build.yml

If I remember correctly in reading some of your messages, you seemed to not have the relevant socketcan libraries and headers installed so you might need to get those as well. Sadly, I don't remember what packages that entails as I haven't broken linux and had to re-install recently.

More or less you want the very latest QT5 build (5.15) and it shouldn't require much in the way of extra plugins except qserialbus and qserialport.

TL;DR - It does use the most recent QT5 but QT6 has been out a LONG time and it hasn't transitioned to needing that so it's not entirely targetted at the bleeding edge. It's just that too many linux distros drag their feet with the QT version so even though QT5.15 is quite old, they still often haven't even updated to that.

**Comment by daveHitchman** [comment]
OK, seems that if you go through the pain and loss of data (all your chrome data is trashed) and upgrade ubuntu to jellyfish it gives you QT 5.15.3 and my bodged version built... I will now undo my bodges and hope it still builds

**Reply by daveHitchman** [reply]
it did :)
